### PR TITLE
Don't allow fs provider to reach above root dir.

### PR DESCRIPTION
**Pull request by mbleigh**
This change makes it so that the fs provider cannot serve files above the specified root. Previously, a string of `../../../` would let you access files outside the specified public directory.

### CODE REVIEW

File: test/integration/serving-files.spec.js [hunk 1]
```diff
@@ -62,6 +63,30 @@ describe('serves', function() {
...
+  });
+
+  it('file with spaces', function(done) {
+    var opts = options();
+
+    var app = connect()
+      .use(superstatic(opts));
+
+    request(app)
+      .get('/dir/')
```
Comment (drtriumph) [inline test/integration/serving-files.spec.js]: Probably missing something, but how does this test a file with spaces? Isn't it just getting the `index.html`?

File: test/integration/serving-files.spec.js [hunk 2]
```diff
@@ -62,6 +63,30 @@ describe('serves', function() {
       .end(done);
   });
 
+  it('cannot access files above the root', function(done) {
+    var app = connect()
+      .use(superstatic(options()));
+
+    request(app)
+      .get('/../README.md')
```
Comment (drtriumph) [inline test/integration/serving-files.spec.js]: Technically, would `/dir/../index.html` work here? Seems a little odd - I've never seen relative paths like this in URLs

File: lib/providers/fs.js [hunk 3]
```diff
@@ -71,7 +78,14 @@ module.exports = function(options) {
...
     var fullPathnames = publicPaths.map(function(p) {
-      return pathjoin(cwd, p, pathname);
+      var bp = pathjoin(cwd, p);
+      var fp = pathjoin(cwd, p, pathname);
+      if (path.relative(bp, fp).indexOf('..') === 0) {
+        return null;
+      }
+      return fp;
+    }).filter(function(p) {
+      return !!p;
```
Comment (drtriumph) [inline lib/providers/fs.js]: This whole function is quite terse - the short variables names make it slow to read through and verify correctness. Also,  `!!` is one of my least favorite expressions in JavaScript - isn't this basically a null check? Why not use `_.startsWith(relativePath, '..')` and `!_.isNull(p)` to be a little more descriptive - the filesystem lookups are always going to be the slowest part of this method

**Comment by drtriumph** [comment]
A few nits, back to you

**Comment by mbleigh** [comment]
Drastically simplified implementation. I don't think there's any reason to allow parent directory pathing at all.

**Comment by drtriumph** [comment]
LGTM, thanks

**Pull request merged**